### ISSUE TITLE
Unexpected behaviour when processing images using python API.

### ISSUE THREAD

**Issue by TheFoolishPupil** [CMU-Perceptual-Computing-Lab/openpose#1824]
### Issue Summary
I am using the python API (working off example 04_keypoints_from_images) to process frames from a video that I intend to stitch back together after processing. I expect (perhaps wrongly) that the images in the folder are processed in order as this seems to be the case, however there are certain images that appear to be consistently out of order.

#### Context
I am putting this together to provide a demo for a potential product that would detect when someone is, for example touching their face, so **this script is not intended to be used in an actual application but just for generating a demo illustrating the idea**. In light of that, I am interested first and foremost in getting a demo video working nicely, but also interested in hearing what the most correct way of implementing this sort of application would be if it were to be a small device running and creating live alerts off these events.

I am extremely interested in help regarding how to move forward on these points as I would like to use OpenPose for the application but will likely have to focus my attention elsewhere if I can't get a demo video working correctly.  If I can, we will potentially be purchasing a commercial license if we move forward with the application.

Thanks for taking the time to hear me out and have a look at my problem. Really cool project and I look forward to hear back. 👾👾

### Type of Issue
- Help wanted
- Unexpected behaviour

### Code
https://pastebin.com/ezp9GsMG

### Video and Images Illustrating Issue
I have provided the source video as well as the final resulting video, which demonstrates the issue/s I am seeing. What I want to achieve is a near-exact copy of the original but with the skeleton overlaid. You can also see that in addition to out-of-place frames, the frames a jittery.
I have also provided some of the frames for closer inspection as well as to show that this is not an issue occuring when stitching frames back together but happens during the processing. I expect (perhaps wrongly) that these are identical except for the skeleton being drawn. You can see that the third image appears to be out of place for example.

Original source video: https://imgur.com/a/dgC3LxG
Resulting video after processing and stitching frames: https://imgur.com/a/RDnDjUA

Photos before: https://imgur.com/a/KSLAvSf
Photos after processing: https://imgur.com/a/ZHnYmW6

### Your System Configuration
2. **OpenPose version**: Latest from github as of 06/01/21

3. **General configuration**:
    - **Installation mode**: CMake
    - **Operating system**: macOS
    - **Operating system version**:  10.15.7
    - **Release or Debug mode**? (by default: release): Not sure if the applies to python API issues, otherwise unsure.
    - Compiler: Apple clang version 12.0.0 (clang-1200.0.32.28)

4. **Non-default settings**:
    - **3-D Reconstruction module added**?: no
    - Any other custom CMake configuration with respect to the default version?: Python built, gpu mode: OpenCL

5. **3rd-party software**:
    - **Caffe version**: Installed as per guide for macOS installation
    - **CMake version**: 3.19.2
    - **OpenCV version**: 4.5.0

8. If **Python** API:
    - **Python version**: 3.7.5
    - **Numpy version** : 1.19.4

**Comment by TheFoolishPupil** [comment]
I managed to solve the issue. The culprit was https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/5a9acc730915f2171badcf10076aef9213f38e01/examples/tutorial_api_python/04_keypoints_from_images.py#L62

Turns out the issue was not in processing but in the image paths. The list of paths returned from the line above are not exactly in order, easily fixed by sorting the list. 

I noticed that this same line throws a segmentation fault when the number of images is sufficiently large (300 in my case). I am able to make it process at least 300 with `imagePaths = sorted([str(filepath) for filepath in pathlib.Path(args[0].image_dir).glob('**/*')])`.
